### PR TITLE
Support for 3d audio

### PR DESCRIPTION
**Pull request by kattalainen**

### CODE REVIEW

File: StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.cpp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #include "stdafx.h"
 #include "AudioVoice.h"
 #include "AudioBuffer.h"
-#include "AudioEngine.h"
+//#include "AudioEngine.h"
```
Comment (Whixx) [inline StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.cpp]: Remove if this serves no purpose

File: StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.h [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #ifndef AUDIOVOICE_H
 #define AUDIOVOICE_H
 #include <xaudio2.h>
+//#include <x3daudio.h>
```
Comment (Whixx) [inline StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.h]: Remove

**Review by Whixx — CHANGES_REQUESTED** [review]
Remove unnessecary comments and add some more comments describing steps in the code.

For example: "This for-loop goes through the sounds and checks if they are active"

**Review by Whixx — DISMISSED** [review]
Make sure to look into the DPSSettings problem

**Comment by kattalainen** [comment]
3d audio now supports multiple sources. Same source can be used for several entities, for both 3d and 2d components at the same time.

File: StortSpelProjekt/Engine/src/ECS/Components/Audio3DEmitterComponent.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+void component::Audio3DEmitterComponent::Play(const std::wstring& name)
+{
+#ifdef _DEBUG
+	if (m_VoiceEmitterData.count(name) != 0)
+	{
+		m_VoiceEmitterData[name].voice.Play();
+	}
+	else
+	{
+		Log::PrintSeverity(Log::Severity::WARNING, "Tried to play Audio that has not been loaded!");
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/ECS/Components/Audio3DEmitterComponent.cpp]: Would be great if the name of the sound was printed as well!
Also, don't forget to add a linebreak at the end of the warning.

File: StortSpelProjekt/Engine/src/ECS/Components/Audio3DEmitterComponent.cpp [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+void component::Audio3DEmitterComponent::Stop(const std::wstring& name)
+{
+#ifdef _DEBUG
+	if (m_VoiceEmitterData.count(name) != 0)
+	{
+		m_VoiceEmitterData[name].voice.Stop();
+	}
+	else
+	{
+		Log::PrintSeverity(Log::Severity::WARNING, "Tried to stop Audio that has not been loaded!");
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/ECS/Components/Audio3DEmitterComponent.cpp]: print the name and add line ending.

File: StortSpelProjekt/Engine/src/ECS/Components/AudioVoiceComponent.cpp [hunk 5]
```diff
@@ -51,6 +58,6 @@ void component::AudioVoiceComponent::Stop(const std::wstring& name)
 		Log::PrintSeverity(Log::Severity::WARNING, "Tried to stop Audio that has not been loaded!");
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/ECS/Components/AudioVoiceComponent.cpp]: Not your code, but would be nice if you could add the name of the sound and a line ending to this print.

**Review by Schtekt — DISMISSED** [review]
Just a few nitpicks with prints.
Otherwise, I see no faults in this (Though I am not well versed in audio...)

**Review by Whixx — DISMISSED** [review]
The audio is working fine and i couldn't find any coding errors. Good work.

**Review by Schtekt — APPROVED** [review]

**Review by Whixx — APPROVED** [review]

**Pull request merged**